enclave/core/sgx/atexit.c

```diff
@@ -41,9 +41,10 @@ static oe_spinlock_t _spin = OE_SPINLOCK_INITIALIZER;
 
 static oe_atexit_entry_t* _new_atexit_entry(void (*func)(void*), void* arg)
 {
+    extern void* _oe_malloc(size_t);
     oe_atexit_entry_t* entry;
 
-    if ((entry = (oe_atexit_entry_t*)oe_sbrk(sizeof(oe_atexit_entry_t))) ==
+    if ((entry = (oe_atexit_entry_t*)_oe_malloc(sizeof(oe_atexit_entry_t))) ==
```

Comment:
Is this safe? Is there a guarantee that _oe_malloc will not rely on atexit/order?

Reply:
So the allocators I know about are very careful not to depend on anything to do with initialisation order.  jemalloc and dlmalloc both on every call check that they are initialised.  I don't know of any allocators that call atexit. Some call a pthread at exit during the per thread initialisation, but again have to be very careful about ensuring the allocator is sufficiently initialised, before making this call.